projects/plugins/jetpack/extensions/blocks/premium-content/_inc/subscription-service/class-token-subscription-service.php

```diff
@@ -71,43 +71,43 @@ public function visitor_can_view_content( $valid_plan_ids, $access_level ) {
 		}
 
 		if ( $is_valid_token ) {
-			if ( 'subscribers' === $access_level ) {
-				return 'active' === $payload['blog_sub'];
-			}
-			$subscriptions = (array) $payload['subscriptions'];
-		} elseif ( is_user_logged_in() ) {
-			/*
-			 * If there is no token, but the user is logged in,
-			 * get current subscriptions and determine if the user has
-			 * a valid subscription to match the plan ID.
-			 */
+			$is_blog_subscrber  = 'active' === $payload['blog_sub'];
+			$subscriptions      = (array) $payload['subscriptions'];
+			$is_paid_subscriber = $this->validate_subscriptions( $valid_plan_ids, $subscriptions );
+		} else {
+			// Token not valid. We bail even unless the post can be accessed publicly.
+			return 'everybody' === $access_level;
```

Comment:
Still not exactly sure about that. What is the use case for the token not to be valid...

Reply:
1. If user is logged in AND can edit post, than also unlock (I forgot about the use case)
2. If access_level is empty, than also unlock

Reply:
I think you can just don't have this check here and forward to user_has_access( $access_level, false, false )

Reply:
But you will need to put `current_user_can('edit_post')`

Reply:
Done